x/staking/keeper/historical_info.go

```diff
@@ -48,7 +50,14 @@ func (k Keeper) TrackHistoricalInfo(ctx context.Context) error {
 		return err
 	}
 
-	historicalEntry := types.NewHistoricalInfo(sdkCtx.BlockHeader(), types.Validators{Validators: lastVals, ValidatorCodec: k.validatorAddressCodec}, k.PowerReduction(ctx))
+	h := cmtproto.Header{
```

Comment:
Looks like app hash is missing here? Should be available in `sdkCtx.HeaderInfo().AppHash`, right?